devops/infrastructure/kubernetes/index.ts

```diff
@@ -0,0 +1,128 @@
+import * as awsx from '@pulumi/awsx'
+import * as eks from '@pulumi/eks'
+import * as k8s from '@pulumi/kubernetes'
+import * as pulumi from '@pulumi/pulumi'
+
+const awsConfig = new pulumi.Config('aws')
+
+// Create a VPC for our cluster.
+const vpc = new awsx.ec2.Vpc('vpc', { numberOfAvailabilityZones: 2 })
+
+// Create an EKS cluster with the default configuration.
+const cluster = new eks.Cluster('eksctl-my-cluster', {
+  vpcId: vpc.id,
+  subnetIds: vpc.publicSubnetIds,
+  instanceType: 't2.micro',
+  providerCredentialOpts: {
+    profileName: awsConfig.get('profile'),
+  },
+})
+
+// Export the cluster's kubeconfig.
+export const kubeconfig = cluster.kubeconfig
+
+// Create a repository
+const repo = new awsx.ecr.Repository('my-repo')
+
+// Build an image and publish it to our ECR repository.
+
+export const colossusImage = repo.buildAndPushImage({
+  dockerfile: '../../../colossus.Dockerfile',
+  context: '../../../',
+})
+
+const name = 'storage-node'
+
+// Create a Kubernetes Namespace
+const ns = new k8s.core.v1.Namespace(name, {}, { provider: cluster.provider })
+
+// Export the Namespace name
+export const namespaceName = ns.metadata.name
+
+// Create a NGINX Deployment
+const appLabels = { appClass: name }
+const deployment = new k8s.apps.v1.Deployment(
+  name,
+  {
+    metadata: {
+      namespace: namespaceName,
+      labels: appLabels,
+    },
+    spec: {
+      replicas: 1,
+      selector: { matchLabels: appLabels },
+      template: {
+        metadata: {
+          labels: appLabels,
+        },
+        spec: {
+          containers: [
+            {
+              name: 'ipfs',
+              image: 'ipfs/go-ipfs:latest',
+              ports: [{ containerPort: 5001 }, { containerPort: 8080 }],
+            },
+            {
+              name: 'colossus',
+              image: colossusImage,
+              env: [
+                {
+                  name: 'WS_PROVIDER_ENDPOINT_URI',
+                  // example 'wss://18.209.241.63.nip.io/'
+                  value: process.env.WS_PROVIDER_ENDPOINT_URI,
+                },
+                {
+                  name: 'DEBUG',
+                  value: '*',
+                },
+              ],
+              command: [
+                'yarn',
+                'colossus',
+                '--dev',
```

Comment:
```suggestion
                '--anonymous',
```

Works better for test setup. This will just run a replicator node.

But ideally we want to be able to provide arguments:
```
    --provider-id ID, -i ID     StorageProviderId assigned to you in working group.
    --key-file FILE             JSON key export file to use as the storage provider (role account).
    --public-url=URL, -u URL    API Public URL to announce.
```

We either need to find a way securely mount a volume to provide the keyfile, or modify colossus to take the raw key from an env variable.

The public url will also either need to be known ahead of time or we can use `nip.io`

Reply:
Added the optional parameters